File: examples/java/spring-boot/one_consumer_two_providers/account-provider/src/main/java/br/com/zup/pact/accountprovider/service/AccountService.java
package br.com.zup.pact.accountprovider.service;

import br.com.zup.pact.accountprovider.dto.AccountDetailsDTO;
import br.com.zup.pact.accountprovider.dto.BalanceDTO;
import java.util.List;
import java.util.Optional;

public interface AccountService {
    Optional<AccountDetailsDTO> getAccountDetailsByClientId(Integer clientId);

    Optional<List<AccountDetailsDTO>> getAll();

    Optional<BalanceDTO> getBalanceByClientId(Integer clientId);
}

File: examples/kotlin/one-consumer-one-provider/provider/src/main/kotlin/br/com/zup/pact/provider/service/AccountService.kt
package br.com.zup.pact.provider.service

import br.com.zup.pact.provider.dto.AccountDetailsDTO
import br.com.zup.pact.provider.dto.BalanceDTO

interface AccountService {
    fun getAccountDetailsByClientId(clientId: Int): AccountDetailsDTO?
    fun getAll(): List<AccountDetailsDTO>
    fun getBalanceByClientId(clientId: Int): BalanceDTO?
}
